src/lsptoolshost/dotnetRuntimeExtensionResolver.ts

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { HostExecutableInformation } from "../shared/constants/HostExecutableInformation";
+import { IHostExecutableResolver } from "../shared/constants/IHostExecutableResolver";
+import { getServerPath } from "../lsptoolshost/roslynLanguageServer";
+import { PlatformInformation } from "../shared/platform";
+import { Options } from "../shared/options";
+import { existsSync } from 'fs';
+import { CSharpExtensionId } from '../constants/CSharpExtensionId';
+
+export const DotNetRuntimeVersion = '7.0';
+
+interface IDotnetAcquireResult {
+    dotnetPath: string;
+}
+
+/**
+ * Resolves the dotnet runtime from given options and the dotnet runtime VSCode extension.
+ */
+export class DotnetRuntimeExtensionResolver implements IHostExecutableResolver {
+    constructor(private platformInfo: PlatformInformation) { }
+
+    private hostInfo: HostExecutableInformation | undefined;
+
+    async getHostExecutableInfo(options: Options): Promise<HostExecutableInformation> {
+        let dotnetRuntimePath = options.commonOptions.dotnetPath;
+        const serverPath = getServerPath(options, this.platformInfo);
```

Comment:
This seems wrong in the current form. This is a generic .NET resolver but it's calling into roslyn's specific `getServerPath` method. Shouldn't we make this fully generic and have the paths / file namse passed in so we can share with razor?

Reply:
Yeah - when I originally wrote it, I hadn't thought about sharing it with Razor yet.  I'll update this.